### PR TITLE
grin-wallet 2.1.0 (new formula)

### PR DESCRIPTION
**Pull request by quentinlesceller**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

**Comment by quentinlesceller** [comment]
Hi @SMillerDev I just addressed this in the latest commit.

File: Formula/grin-wallet.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class GrinWallet < Formula
+  desc "Grin Wallet"
```
Comment (zachauten) [inline Formula/grin-wallet.rb]: Could you add more info about what the app does beyond just the name?

**Review by zachauten — CHANGES_REQUESTED** [review]
Please squash the branch down to a single commit.

File: Formula/grin-wallet.rb (hunk 1 above)
Comment (quentinlesceller) [reply Formula/grin-wallet.rb]: I've added the following:
```
"Grin Wallet, official wallet for the cryptocurrency Grin"
```

**Comment by quentinlesceller** [comment]
@zachauten Thanks. Squashed to a single commit.

File: Formula/grin-wallet.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class GrinWallet < Formula
+  desc "Grin Wallet, official wallet for the cryptocurrency Grin"
```
Comment (zbeekman) [inline Formula/grin-wallet.rb]: How about:
```suggestion
  desc "Official wallet for the cryptocurrency Grin"
```

Otherwise it's repetitive

**Review by zbeekman — CHANGES_REQUESTED** [review]
If you agree with my suggestion, can you please make the change and squash/fixup the commit? Should be ready to go after that.

**Comment by quentinlesceller** [comment]
Agree with the suggestion. Changed and squashed.

File: Formula/grin-wallet.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class GrinWallet < Formula
+  desc "Official wallet for the cryptocurrency Grin"
+  homepage "https://grin.mw"
+  url "https://github.com/mimblewimble/grin-wallet/archive/v2.1.0.tar.gz"
+  sha256 "649664af929fc2afd42e52597ad675d3e1856c95e0cda60ef8d4c478abe5c01c"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--root", prefix, "--path", "."
```
Comment (zbeekman) [inline Formula/grin-wallet.rb]: Sorry one more thing: Do you have a cargo.lock file? If so can you please add `--lock` to the install? If grinwallet doesn't have one, what would it take to convince upstream to add one?

**Review by zbeekman — CHANGES_REQUESTED** [review]
LGTM, but is it possible to add `--lock` to the cargo install command? Does upstream provide a `cargo.lock` file?

File: Formula/grin-wallet.rb (hunk 3 above)
Comment (quentinlesceller) [reply Formula/grin-wallet.rb]: We do. Adding the `--locked` flag.

Comment (zbeekman) [reply Formula/grin-wallet.rb]: Thanks! Glad you didn't add `--lock` instead of `--locked` too!

**Review by zbeekman — APPROVED** [review]
LGTM

**Review by chenrui333 — APPROVED** [review]

**Review by zachauten — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks all, merging now!

**Pull request closed without merging**